### PR TITLE
Simplified Excemption Names

### PR DESCRIPTION
**Pull request by caughtquick**
Agreed on in the discord to be simplified

### CODE REVIEW

**Comment by FjorgeTheHuman** [comment]
tbh they sound funny now

**Comment by thewarsawpakt** [comment]
Not a fan of the new names. The old ones were verbose for a reason, and these use uncommon shortenings of words which make it confusing to read.

**Comment by caughtquick** [comment]
then how do you expect me to shorten it?

**Comment by FjorgeTheHuman** [comment]
~~by using `print()` instead of exceptions~~

**Comment by caughtquick** [comment]
not a solution @FjorgeTheHuman, its been decided we are using exceptions

**Comment by FjorgeTheHuman** [comment]
> not a solution @FjorgeTheHuman, its been decided we are using exceptions

it has not been decided

**Comment by thewarsawpakt** [comment]
Look into `sys.excepthook`.

**Comment by caughtquick** [comment]
> Look into `sys.excepthook`.

Me?

**Review by Sophon96 — CHANGES_REQUESTED** [review]
Nope, this is worse than being too verbose. Its hard to tell what the error means and its awkward.

**Comment by thewarsawpakt** [comment]
> > Look into `sys.excepthook`.
> 
> Me?

Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.

**Comment by FjorgeTheHuman** [comment]
> > > Look into `sys.excepthook`.
> > 
> > 
> > Me?
> 
> Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.

i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh

**Comment by caughtquick** [comment]
> Nope, this is worse than being too verbose. Its hard to tell what the error means and its awkward.

its not great, the others gave more detail, but if we really wanted to shorten it I guess this is the only real way

**Comment by thewarsawpakt** [comment]
> > > > Look into `sys.excepthook`.
> > > 
> > > 
> > > Me?
> > 
> > 
> > Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.
> 
> i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh

That's what `sys.excepthook` does, check the discord. If we use the sys function, I see no reason for this change.

**Comment by FjorgeTheHuman** [comment]
and also a way to control the exit code when exiting

**Comment by Sophon96** [comment]
> and also a way to control the exit code when exiting

`exit(<exit code>)`

**Comment by thewarsawpakt** [comment]
> and also a way to control the exit code when exiting

When would this be useful? If it errors then we exit with 1, if it's successful we exit with a 0. Exit code doesn't really matter in my opinion.

**Comment by FjorgeTheHuman** [comment]
> > > > > Look into `sys.excepthook`.
> > > > 
> > > > 
> > > > Me?
> > > 
> > > 
> > > Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.
> > 
> > 
> > i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh
> 
> That's what `sys.excepthook` does, check the discord. If we use the sys function, I see no reason for this change.

the long error type is also a pain to type out during the code and usually goes past the width of my screen making writing stuff harder still, so thats another reason id like to shorten them

File: LISENCE [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (Sophon96) [inline LISENCE:1]: Bruh wrong file name

**Comment by thewarsawpakt** [comment]
> > > > > > Look into `sys.excepthook`.
> > > > > 
> > > > > 
> > > > > Me?
> > > > 
> > > > 
> > > > Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.
> > > 
> > > 
> > > i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh
> > 
> > 
> > That's what `sys.excepthook` does, check the discord. If we use the sys function, I see no reason for this change.
> 
> the long error type is also a pain to type out during the code and usually goes past the width of my screen making writing stuff harder still, so thats another reason id like to shorten them

how small is your screen, also autocomplete

**Review by Sophon96 — CHANGES_REQUESTED** [review]
Bruh wrong file name

File: LISENCE (hunk 1 above)
Comment (thewarsawpakt) [reply LISENCE]: -_- Who did this

**Comment by FjorgeTheHuman** [comment]
> > and also a way to control the exit code when exiting
> 
> When would this be useful? If it errors then we exit with 1, if it's successful we exit with a 0. Exit code doesn't really matter in my opinion.

I think it'd be useful still you never know so I just wanna be prepared

**Comment by FjorgeTheHuman** [comment]
> > > > > > > Look into `sys.excepthook`.
> > > > > > 
> > > > > > 
> > > > > > Me?
> > > > > 
> > > > > 
> > > > > Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.
> > > > 
> > > > 
> > > > i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh
> > > 
> > > 
> > > That's what `sys.excepthook` does, check the discord. If we use the sys function, I see no reason for this change.
> > 
> > 
> > the long error type is also a pain to type out during the code and usually goes past the width of my screen making writing stuff harder still, so thats another reason id like to shorten them
> 
> how small is your screen, also autocomplete

my resolution is `1920x1080`

File: LISENCE (hunk 1 above)
Comment (caughtquick) [reply LISENCE]: ??

**Comment by thewarsawpakt** [comment]
> > > and also a way to control the exit code when exiting
> > 
> > 
> > When would this be useful? If it errors then we exit with 1, if it's successful we exit with a 0. Exit code doesn't really matter in my opinion.
> 
> I think it'd be useful still you never know so I just wanna be prepared

If there isn't a good example as to why we would use or need it, then I don't think it should be considered. Most shells don't even show the exit code, therefore having a custom one is next to useless.

**Comment by Sophon96** [comment]
> > > > > > > > Look into `sys.excepthook`.
> > > > > > > 
> > > > > > > 
> > > > > > > Me?
> > > > > > 
> > > > > > 
> > > > > > Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.
> > > > > 
> > > > > 
> > > > > i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh
> > > > 
> > > > 
> > > > That's what `sys.excepthook` does, check the discord. If we use the sys function, I see no reason for this change.
> > > 
> > > 
> > > the long error type is also a pain to type out during the code and usually goes past the width of my screen making writing stuff harder still, so thats another reason id like to shorten them
> > 
> > 
> > how small is your screen, also autocomplete
> 
> my resolution is `1920x1080`

Semantic completion to the rescue

**Comment by FjorgeTheHuman** [comment]
> > > > > > > > > Look into `sys.excepthook`.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Me?
> > > > > > > 
> > > > > > > 
> > > > > > > Fjorge. This is a compromise which allows us to use both print and traceback, handled by a predefined function.
> > > > > > 
> > > > > > 
> > > > > > i'd be okay with exceptions as long as u find out a way to cut out the exception type part from the error message tbh
> > > > > 
> > > > > 
> > > > > That's what `sys.excepthook` does, check the discord. If we use the sys function, I see no reason for this change.
> > > > 
> > > > 
> > > > the long error type is also a pain to type out during the code and usually goes past the width of my screen making writing stuff harder still, so thats another reason id like to shorten them
> > > 
> > > 
> > > how small is your screen, also autocomplete
> > 
> > 
> > my resolution is `1920x1080`
> 
> Semantic completion to the rescue

not everywhere can I get semantic completion

File: LISENCE (hunk 1 above)
Comment (caughtquick) [reply LISENCE]: oi i tried to create a lisence file on the main branch. yeeted now

**Comment by FjorgeTheHuman** [comment]
ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish

**Comment by thewarsawpakt** [comment]
> ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish

https://github.com/Generic-Linux/libtwirl/pull/18#issuecomment-790915694

**Comment by caughtquick** [comment]
> > ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish
> 
> [#18 (comment)](https://github.com/Generic-Linux/libtwirl/pull/18#issuecomment-790915694)

github discussions we go

**Comment by FjorgeTheHuman** [comment]
> > > ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish
> > 
> > 
> > [#18 (comment)](https://github.com/Generic-Linux/libtwirl/pull/18#issuecomment-790915694)
> 
> github discussions we go

no need the convo is basically over, I'm pretty fine with these names especially if we're not gonna expose them to the user

**Review by FjorgeTheHuman — APPROVED** [review]
Short names cool

**Comment by thewarsawpakt** [comment]
> > > > ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish
> > > 
> > > 
> > > [#18 (comment)](https://github.com/Generic-Linux/libtwirl/pull/18#issuecomment-790915694)
> > 
> > 
> > github discussions we go
> 
> no need the convo is basically over, I'm pretty fine with these names especially if we're not gonna expose them to the user

Which names? also @caughtquick you aren't using a different branch for each feature are you?

**Comment by FjorgeTheHuman** [comment]
by default anyways

> > > > > ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish
> > > > 
> > > > 
> > > > [#18 (comment)](https://github.com/Generic-Linux/libtwirl/pull/18#issuecomment-790915694)
> > > 
> > > 
> > > github discussions we go
> > 
> > 
> > no need the convo is basically over, I'm pretty fine with these names especially if we're not gonna expose them to the user
> 
> Which names? also @caughtquick you aren't using a different branch for each feature are you?

the shortened error names, if we aren't exposing them to the user by default its good enough

**Comment by caughtquick** [comment]
> > > > > ok can we stop with the quoting now we're gonna end up having 20 layers of quotes by the time we finish
> > > > 
> > > > 
> > > > [#18 (comment)](https://github.com/Generic-Linux/libtwirl/pull/18#issuecomment-790915694)
> > > 
> > > 
> > > github discussions we go
> > 
> > 
> > no need the convo is basically over, I'm pretty fine with these names especially if we're not gonna expose them to the user
> 
> Which names? also @caughtquick you aren't using a different branch for each feature are you?

im using a different branch of features on my repo, i just screwed up this time and didn't, my bad

**Comment by thewarsawpakt** [comment]
> the shortened error names, if we aren't exposing them to the user by default its good enough

These shortened error names make it hard to write because I don't know wtf they mean. If we can find a compromise between shorter names and readable ones, I'd be willing to merge this.

**Comment by FjorgeTheHuman** [comment]
ok I'm gonna write a new errors PR with hopefully better short names and all that sys magic then

**Pull request closed without merging**